src/Inventory/Asset/NetworkPort.php

```diff
@@ -250,8 +250,25 @@ private function handleLLDPConnection(\stdClass $port, int $netports_id)
             $rule->getCollectionPart();
             $rule->processAllRules($input, [], ['class' => $this]);
 
-            if (count($this->connection_ports)) {
-                $connections_id = current(current($this->connection_ports));
+            if (count($this->connection_ports) != 1) {
+                continue;
+            }
+
+            $connection_ports = current($this->connection_ports);
+            if (count($connection_ports) == 1) { // single NetworkPort
+                $connections_id = current($connection_ports);
+            } else { // multiple NetworkPorts
+                $networkPort = new \NetworkPort();
+                foreach (array_keys($connection_ports) as $k) {
+                    $networkPort->getFromDB($k);
+                    if ($networkPort->fields['logical_number'] > 0) {
```

Comment:
```suggestion
                    if ($networkPort->getFromDB($k) && $networkPort->fields['logical_number'] > 0) {
```

Reply:
This is also redundant. `$k` from `array_keys($connection_ports)` has to be a valid NetworkPort id.